### ISSUE TITLE
Create new page without dialog asking for title.

### ISSUE THREAD

**Issue by mworrell** [zotonic/zotonic#1044]
If a new resource is created we first ask the title of that resource in a dialog.

Question is if we can skip that dialog and go directly to a form for editing the new resource.
Problem is that users might click on the *new page* button and create many empty resources.

Proposed solution is to create a temporary new resource, the created resource will be coupled to the page process. If the page process stops, then the newly created resource will be deleted iff the title is empty and the resource is (still) unpublished.

Together with the empty resource we can also insert a pivot task for deleting the resource. This pivot task can check if the creating page process exists (by id), if not then the resource can be deleted.

The resource can be created with a filter, providing all the necessary initial properties:

```
{% with id|temporary_rsc:{props category=... foo=bar  qargs} as id %}
   ... show edit form ...
{% endwith %}
```

By making a new temporary resource we can keep all edit routines unchanged and prevent handling the case of a non-existant in uploads, edges, etc.

The initial properties can also be added from the query args, by means of the *qargs* option above. This enables dispatch rules like `/new/article`.

/cc @fredpook @ddeboer @ArthurClemens @DorienD

**Comment by mworrell** [comment]
This filter could be added to `mod_admin`

**Comment by arjan** [comment]
I really like the dialog. What's the use case for removing it?
Your proposal seems like a lot of work and administration..

Also for creating resources from the page connections dialog it is still needed.

**Comment by mworrell** [comment]
It is actually quite easy, this just introduces "dummy" resources which will be deleted if not saved.

This is for a use case in front ends where users can add content via simple means.

**Comment by ArthurClemens** [comment]
So if this mechanism is in place, we can keep the dialogs?

**Comment by fredpook** [comment]
If you don't change your frontend UX the dialogs stay as before

**Comment by mworrell** [comment]
Indeed the UX stays the same, this is just an extra thing that is useful for simplifying the flow when creating new documents in some specific front ends.

**Comment by arjan** [comment]
In that case, :+1: when the admin UI stays as it is now.

The only thing is that this temporary resource thing will create "gaps" in the rsc id sequence, when temporary resources get deleted again. Like if you just reload the "new rsc" page 5 times the rsc seq counter will be increased by 5, or not?

**Comment by mworrell** [comment]
Yes, it will. We might want to add a search for recently added resources by the same user and then just use the last one (if the category is right). That prevents the problem where somebody makes a bot adding pages in bulk...

**Comment by mworrell** [comment]
This bulk-insert-prevention can be implemented by registering the watch process using a combination of the session-id and category of the to be inserted resource.  The deletion can then be done with a timeout, so that the speed of the bulk insert can be slowed to an acceptable rate.

Other option is to use counters and a tarpit, which might also be interesting for other bulk and/or DOS requests.

**Comment by mworrell** [comment]
Moved to the 0.13.6 release, after consultation with @fredpook

**Comment by ddeboer** [comment]
Release early and often, so :+1:!

**Comment by mworrell** [comment]
Was almost finished, so included in the 0.13.5.
